### PR TITLE
[PVR] Direct channel number input improvement

### PR DESCRIPTION
**Pull request by ksooo**
By pressing "OK" (Enter) you can now shortcut the direct channel number input delay and trigger the associated action immediately. Formerly, you had to wait for the internal timeout after that the action gets executed automatically. Ofc, this does still work.

(Last commit is unrelated, just something random I came across by luck, and fixes just a log text.)

@Jalle19 good to go?

### CODE REVIEW

**Comment by Jalle19** [comment]
I thought this worked already? Or is this fealing with a v18 regression?

Code looks cleaner in any case, +1

**Comment by ksooo** [comment]
> I thought this worked already? 

 No, it didn't. ;-)

> Or is this fealing with a v18 regression?

 Well, it was 'not yet implemented' for 18. 

 In 17 and before, where we used a numeric input dialog, you could ofc. end the dialog with OK and then the action was executed immediately. So, yes, it's a regression from user's point of view.

**Comment by Jalle19** [comment]
I see, in that case +1

File: xbmc/pvr/windows/GUIWindowPVRGuide.cpp [hunk 1]
```diff
@@ -335,6 +335,17 @@ bool CGUIWindowPVRGuideBase::OnMessage(CGUIMessage& message)
     {
       if (message.GetSenderId() == m_viewControl.GetCurrentControl())
       {
+        if (message.GetParam1() == ACTION_SELECT_ITEM ||
```
Comment (Jalle19) [inline xbmc/pvr/windows/GUIWindowPVRGuide.cpp]: Could this be handled in the base class so the code wouldn't have to be duplicated?

Comment (ksooo) [reply xbmc/pvr/windows/GUIWindowPVRGuide.cpp]: I suggest to keep it as it is, as only two out of five pvr window classes need this.

**Comment by ksooo** [comment]
jenkins build this please

File: xbmc/pvr/windows/GUIWindowPVRGuide.cpp (hunk 1 above)
Comment (Jalle19) [reply xbmc/pvr/windows/GUIWindowPVRGuide.cpp]: Sure

**Review by Jalle19 — APPROVED** [review]

**Pull request merged**